src/vs/workbench/contrib/terminalContrib/suggest/browser/terminalSuggestAddon.ts

```diff
@@ -847,3 +917,31 @@ export function normalizePathSeparator(path: string, sep: string): string {
 	}
 	return path.replaceAll('/', '\\');
 }
+
+function inputMostlyMatchesLabel(commandLine: string, label: string): boolean {
+	const lastWordOfCommandLine = commandLine.split(' ').pop() ?? '';
+	let labelIndex = 0;
+	let matchedChars = 0;
+	// At least half of label must be matched
+	const requiredMatches = Math.ceil(label.length / 2);
+
+	for (const char of lastWordOfCommandLine) {
+		if (char === label[labelIndex]) {
+			matchedChars++;
+			labelIndex++;
+		}
```

Comment:
I think this is checking whether the first half of the completion in intact but only for the last word in the command line? So it would always fail if the label contained a space?

Not perfect again, but we could simplify the tracking of this and just check if `commandLine.includes(label.substring(0, Math.ceil(label.length / 2))`? The function name could then be very clear: `inputContainsFirstHalfOfLabel`.

Reply:
That's what I had originally and changed it to be more sophisticated... but yeah, seems like that's fine since the first part of it being present is more important anyway.